### ISSUE TITLE
Rule Request: directory_name_no_space prevent whitespace characters in directory names

### ISSUE THREAD

**Issue by timcmiller** [realm/SwiftLint#3012]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Prevent any whitespace characters in directory names. Having spaces in directories cause issues (or annoyances) when trying to get a path to files located inside the directory since you have to escape the space.

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

There is ambiguity on folder naming between different cases, however it it standard convention to not include spaces in folder names. See [Swift](https://github.com/apple/swift) which uses either kebab-case or camelCase for folder names. See this project which uses both camelCase and snake_case.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Does not trigger
```
DirectoryName
Directory_Name
Directory-Name
```

Does Trigger
```
Directory Name
```

3. Should the rule be configurable, if so what parameters should be configurable?

With the exception of an exclusion list for folders you might want to exclude (See Supporting Files for this project), no. 

4. Should the rule be opt-in or enabled by default? Why?

Yes, because Apple doesn't have a defined convention for naming folders.

**Comment by jpsim** [comment]
Any reason not to combine this with the file name no space rule?

**Comment by timcmiller** [comment]
> Any reason not to combine this with the file name no space rule?

While it's commonly accepted practice across projects, there is much less supporting documented standards that specifically call our folder names not having spaces. With a lack of supporting evidence I thought it made sense to have two separate rules to allow levels of opting in.

**Comment by jpsim** [comment]
I think that's fine. The alternative would be to make this a configurable option on a combined `no_spaces_in_paths` rule, but this accomplishes the same effect.
